pkg/distributor/push.go

```diff
@@ -51,9 +55,10 @@ func Handler(
 	sourceIPs *middleware.SourceIPExtractor,
 	allowSkipLabelNameValidation bool,
 	limits *validation.Overrides,
+	retryCfg *RetryConfig,
```

Comment:
question from c++ user. and for my learning, In golang is that really optimised in most of the case pointer or object become unimportant? I always pass object by pointer taking assumption it would be no copy and more efficient.

Reply:
or we simply say, whenever it is not optional, go with the copy.